### ISSUE TITLE
Uncaught TypeError: Cannot call method 'open' of undefined

### ISSUE THREAD

**Issue by martinloeffen** [pwlin/cordova-plugin-file-opener2#143]
For some strange and unknown reason, I receive an error message when calling `cordova.plugins.fileOpener2.open( '/test.pdf', 'application/pdf');` namely Cannot call method 'open' of undefined

I have been trying to track this down for a while. Everything point at that it should be working. See my Stackexchange question. https://stackoverflow.com/questions/44757919/phonegap-plugins-dont-initialise

I have the following code, that will always `alert("Undefined")`

```javascript
document.addEventListener("deviceready", function() {
            alert("open pdf fileopener2");
            if (typeof cordova.plugins.fileOpener2 !== "undefined") {
                cordova.plugins.fileOpener2.open( '/test.pdf', 'application/pdf');
            }
            else {
                alert("Undefined");
            }
        });
```

I have the same issue with another file opener plugin, but not with other plugins (barcode scanner, social sharing).

There is a same issue with the same title on this tracker, and your reply is that it is presumably fixed when the user updated their Cordova(?). My cordova.version returns 4.4.0, which is the latest, I think. I use PhoneGap desktop.

cordova.plugins contains only barcodeScanner and screenorientation

**Comment by pwlin** [comment]
I think there is something wrong with how you setup your stuff. 

Test with a brand new Cordova project and only add this plugin and see if that works.

**Comment by martinloeffen** [comment]
OK, here are my steps:

* PhoneGap desktop: Create new PhoneGap project
* Use the Hello World template
* Edited config.xml, added `<plugin name="cordova-plugin-file-opener2" source="npm" />` below `<plugin name="cordova-plugin-file-transfer" source="npm" spec="~1.5.0" />`
* Edited index.js:

```javascript
onDeviceReady: function() {
        app.receivedEvent('deviceready');	
		alert(typeof cordova.plugins.fileOpener2);
		cordova.plugins.fileOpener2.open('test.pdf',  'application/pdf');
    }
```

* When the app is opened, the alert says 'undefined' on my three devices (Android, iOS, WinMo)

**Comment by pwlin** [comment]
I am not sure how PhoneGap handles stuff. Try to create a real Cordova project and add this plugin through command line. Here is a simple guide: https://cordova.apache.org/docs/en/latest/guide/cli/index.html

**Comment by pwlin** [comment]
I just noticed that you are trying to open a pdf file without passing the full file path. 

    cordova.plugins.fileOpener2.open('test.pdf',  'application/pdf');

typing `test.pdf` will not work. You need to pass the full file path. Use [Cordova's File plugin](https://cordova.apache.org/docs/en/latest/reference/cordova-plugin-file/).

**Comment by martinloeffen** [comment]
I'll get back to you on Friday. It's giving me another error at this point, when I run `cordova requirements` with just the Android target: Unsupported major.minor version 52.0. 

In the meantime I've tried `cordova plugin remove cordova-plugin-file-opener2` and `cordova plugin add cordova-plugin-file-opener2` within the PhoneGap HelloWorld project. It ran successfully, but it didn't change anything (no difference with just using config.xml).

**Comment by juansaa** [comment]
Hola estoy intentando usar este mismo plugin para android 7 y me da el mismo error.
Te agradezco si me dices si los has solucionado. 

Hi, I'm trying to use this same plugin for android 7 and it gives me the same error.
I thank you if you tell me if you have solved them.
Gracias amigo.
thks.
Juan

**Comment by anonymous-d0z42ujxuog5** [comment]
Hola amigo,

No, I tried everything. Nothing helped. Same issue in pure Cordova. Very frustrating... I ended up using just window.open()

```js
    var url = $('#PDF_ShareURL').val();
    if (!!window.cordova) { // Is Cordova loaded?
        window.open(url, '_system');
    }
    else {
        window.open(url, '_blank');
    }
```

**Comment by juansaa** [comment]
I can use the plugin.
1° in console $npm i cordova pluginname (add plugin a protect)
2° in console $cordova plugin add pluginname (add plugin name to config.xml)


El 29 sept. 2018 9:00 a. m., "anonymous" <notifications@github.com>
escribió:

Hola amigo,

No, I tried everything. Nothing helped. Same issue in pure Cordova. Very
frustrating... I ended up using just window.open()

    var url = $('#PDF_ShareURL').val();
    if (!!window.cordova) { // Is Cordova loaded?
        window.open(url, '_system');
    }
    else {
        window.open(url, '_blank');
    }

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/pwlin/cordova-plugin-file-opener2/issues/143#issuecomment-425639598>,
or mute the thread
